React/Executors/RCTJSCExecutor.m

```diff
@@ -324,6 +324,20 @@ - (void)setUp
                                                  name:event
                                                object:nil];
   }
+
+  // Inject handler used by HMR
+  [self addSynchronousHookWithName:@"nativeInjectHMRUpdate" usingBlock:^(NSString *sourceCode, NSString *sourceCodeURL) {
+    RCTJSCExecutor *strongSelf = weakSelf;
+    if (!strongSelf.valid) {
+      return;
+    }
+
+    JSStringRef execJSString = JSStringCreateWithUTF8CString(sourceCode.UTF8String);
+    JSStringRef jsURL = JSStringCreateWithUTF8CString(sourceCodeURL.UTF8String);
+    JSEvaluateScript(strongSelf->_context.ctx, execJSString, NULL, jsURL, 0, NULL);
```

Comment:
It was partly just moved, but I also made it use the same `addSynchronousHookWithName:usingBlock:` as every thing else that's like it does in the executor. I did this after talking with @martinbigio, who wrote the inject hmr code in the first place, telling me that he didn't realize that method was there. So it's just a bit of cleanup is all.

Regarding actually moving it, from `executeApplicationScript` to `setUp`, where the other native hooks are injected...it just seemed like the correct place for it. But then I remembered that when talking with @martinbigio, he said that the reason he had put the injector in `executeApplicationScript` is because the context evidently gets cleared every time `executeApplicationScript` is called. However, I'm not actually sure if that matters for this use case...I'm pretty sure (based on my read of the code) that there's kind of a 1-to-1 relationship between the executor and the bridge on iOS, and so if a new script is executed, chances are the bridge is going to be torn down and rebuilt, along with the executor. I may be reading that wrong, but that's my current understanding.